pkg/transport/generic_transport_controller.go

```diff
@@ -244,16 +244,30 @@ func (c *genericTransportController) processNextWorkItem(ctx context.Context, wo
 		return false
 	}
 
-	if err := c.process(ctx, obj); err != nil {
-		c.logger.Error(err, "failed to process Binding object", "objectKey", obj, "workerID", workerID)
-	} else {
-		c.logger.Info("synced Binding object successfully.", "objectKey", obj, "workerID", workerID)
-	}
-
+	var err error
+	var retry bool
+	defer func() {
+		if err == nil {
+			// If no error occurs we Forget this item so it does not
+			// get queued again until another change happens.
+			c.workqueue.Forget(obj)
+			c.logger.Info("Synced Binding object successfully.", "objectKey", obj, "workerID", workerID)
+		} else if retry {
+			c.workqueue.AddRateLimited(obj)
+			c.logger.Info("Encountered transient error while processing Binding object, will retry later", "objectKey", obj, "workerID", workerID, "err", err)
```

Comment:
Maybe move to higher log level ?   We normally don't care about these transient errors.

Reply:
I think that there are a lot of logging level adjustments to be made. I think that it will be best to do that after I finish all the plumbing of control over the logging level.

At the current stage of development I recommend consistency with the level of the other Info message about the outcome of `c.process(ctx, obj)`.

Reply:
I'm not sure regular user seeing such error will not still be alarmed, it doesn't directly implied here that it is an expected behavior.